### ISSUE TITLE
line() doesn't work with 1x1-shaped tensor

### ISSUE THREAD

**Issue by Kegnarok** [facebookresearch/visdom#444]
**Bug Description**
The `line` method works with 1-shaped tensors, but not with 1x1-shaped ones (sending one line with one point should be perfectly valid).

**Reproduction Steps**
```python
>>> a
array([[ 1.]])
>>> vis.line(a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/anaconda3/lib/python3.6/site-packages/visdom/__init__.py", line 206, in result
    return fn(*args, **kwargs)
  File "/opt/anaconda3/lib/python3.6/site-packages/visdom/__init__.py", line 769, in line
    update=update, name=name)
  File "/opt/anaconda3/lib/python3.6/site-packages/visdom/__init__.py", line 206, in result
    return fn(*args, **kwargs)
  File "/opt/anaconda3/lib/python3.6/site-packages/visdom/__init__.py", line 635, in scatter
    assert Y.ndim == 1, 'Y should be one-dimensional'
AssertionError: Y should be one-dimensional
```

**Expected behavior**
1x1-tensors should behave exactly as 1-shaped ones.

**Additional context**
Culprit is here: https://github.com/facebookresearch/visdom/blob/master/py/visdom/__init__.py#L1036
(should squeeze only specific dimensions, because the current behavior changes if some dimensions are 1).

**Comment by JackUrb** [comment]
HI @Kegnarok - I'd be happy to accept the fix to this one if you have specific knowledge on the cases that cause np.squeeze to act strangely. Otherwise I'll figure out what the proper squeeze semantics should be in the near future. Thanks for bringing this one up!

**Comment by TomVeniat** [comment]
I think the problem is the same as in https://github.com/facebookresearch/visdom/issues/301:
https://github.com/facebookresearch/visdom/blob/00dd08f5f1e11b56559763c00c0a4db0addcd6a5/py/visdom/__init__.py#L1109
https://github.com/facebookresearch/visdom/blob/00dd08f5f1e11b56559763c00c0a4db0addcd6a5/py/visdom/__init__.py#L1135
Since what we really want is a one-dimensional array, one way to fix this issue would be to flatten the arrays/lists instead of squeezing them. This way we avoid getting a 0-dimensional arrays when there is a single element. 
You mention in https://github.com/facebookresearch/visdom/issues/301 that the squeeze  has a good reason to be here, do you think this solution can still work ? If so I can send a PR.

**Comment by JackUrb** [comment]
@TomVeniat the flatten solution should work correctly, thanks for the catch!

**Comment by JackUrb** [comment]
Fixed in #502, forgot to close.
